presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -829,7 +830,11 @@ else if (isPushedDownSubfield(column)) {
                     columnMappings.add(columnMapping);
                     regularIndex++;
                 }
-                // TODO handle ROW_ID here
+                // ROW_ID is synthesized but not prefilled.
```

Comment:
We can use the prefilled value to pass the partition id component to the reader, and then expect the updated values back. But it will be an ugly design.